pkg/payload/task_graph.go

```diff
@@ -487,9 +492,10 @@ func RunGraph(ctx context.Context, graph *TaskGraph, maxParallelism int, fn func
 	}
 
 	var inflight int
-	nextNode := getNextNode()
-	done := false
+	var done bool
+
 	for !done {
+		nextNode := getNextNode()
```

Comment:
Doing this at the start of the cycle is equivalent to doing it once before the cycle + at the end of each cycle if !done